app/utils/Assets.scala

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import java.net.URL
+import controllers.CachedAssets.hashedPaths
+import play.api.Play
+import org.apache.commons.io.IOUtils
+
+object Assets {
+
+  def inlineResource(path: String) = {
+    val url = AssetFinder("public/" + hashedPaths.getOrElse(path, path))
+    IOUtils.toString(url)
```

Comment:
You should be able to simplify the entire module by using  following snippet 

```scala

import play.api.Play.current
import scala.io.Source

object Asset {
  def inlineResource(path: String): String = 
     current.resource.map(Source.fromURL(_).mkString).getOrElse(
         sys.error(s"Cannot find asset $path. You probably need to run 'grunt compile'."))
}
``` 

References:
[scala.io.Source](http://www.scala-lang.org/api/current/index.html#scala.io.Source$):